### PR TITLE
Conformance test: update getUniform for WebGL 2

### PR DESCRIPTION
**Pull request by Richard-Yunchao**
PTAL when you have time. Thanks a lot!

### CODE REVIEW

File: sdk/tests/conformance/resources/matForWebGL2UniformShader.vert [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+** THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+** EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+** MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+** IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+** CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+** TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+*/
+
+#version 300 es
```
Comment (kenrussell) [inline sdk/tests/conformance/resources/matForWebGL2UniformShader.vert]: The #version directive needs to be the first line of the shader, so the copyright header needs to be moved down.

File: sdk/tests/conformance/resources/samplerForWebGL2UniformShader.frag [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+** THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+** EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+** MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+** IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+** CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+** TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+*/
+
+#version 300 es
```
Comment (kenrussell) [inline sdk/tests/conformance/resources/samplerForWebGL2UniformShader.frag]: Same here.

File: sdk/tests/conformance/resources/uintUniformShader.vert [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+** THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+** EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+** MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+** IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+** CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+** TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+*/
+
+#version 300 es
```
Comment (kenrussell) [inline sdk/tests/conformance/resources/uintUniformShader.vert]: Same here.

File: sdk/tests/conformance/resources/uniformBlockShader.frag [hunk 4]
```diff
@@ -21,6 +21,8 @@
 ** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
 */
 
+#version 300 es
```
Comment (kenrussell) [inline sdk/tests/conformance/resources/uniformBlockShader.frag]: And here.

File: sdk/tests/conformance/resources/uniformBlockShader.vert [hunk 5]
```diff
@@ -21,6 +21,8 @@
 ** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
 */
 
+#version 300 es
```
Comment (kenrussell) [inline sdk/tests/conformance/resources/uniformBlockShader.vert]: Here too.

**Comment by kenrussell** [comment]
Thanks for this contribution. It looks good except for the fact that the #version directive needs to be the first thing in the shaders according to the spec. Could you please fix that up? We'll gladly merge this afterward. Thanks.

**Comment by Richard-Yunchao** [comment]
@kenrussell , thanks for your reviewing. matForWebGL2Program and samplerForWebGL2Program failed to link (getProgramParameter with LINK_STATUS got false) in firefox nightly build in Linux. Could you help to look into the shader carefully? Please help to merge if you think it is OK. You know, firefox nightly build may have errors that lead to this link failure.

**Comment by kenrussell** [comment]
It looks to me like none of the shaders referenced via paths like "../../conformance/resources/boolUniformShader.vert" are loading correctly in Firefox because the web server (at least "python -m SimpleHTTPServer") reports 404s for them. Please debug this test further on your end. Thanks.

**Comment by Richard-Yunchao** [comment]
@kenrussell, It is OK for me. I run "python -m SimpleHTTPServer" under this project, then use firefox to open 127.0.0.1:8000. BTW, I also use apache: put this project under /var/www/webgl in Linux. And access it from localhost:8000/webgl/ or 127.0.0.1:8000/webgl, It is OK too.

But matForWebGL2Program and samplerForWebGL2Program in this patch are failed to compile in firefox nightly build. Other sub tests for getUniform can pass.

**Comment by kenrussell** [comment]
I've had a difficult time testing these new tests -- when running Firefox Nightly on Mac OS X with the WebGL 2.0 prototype enabled *and* the correct URL argument to enable WebGL 2.0 in the test harness, I see many failures. Regardless, the new test basically looks OK and we'll have to get one of the WebGL 2.0 implementations up to a level of quality to try running it to shake out the remaining bugs.

Thanks for this contribution. Merging now.

**Comment by Richard-Yunchao** [comment]
@kenrussell , understood. I also spend some time to run webgl 2.0 test suite on firefox nightly build, including install a new graphics card. And see many failures...

Yeah. I agree with you. It is hard to make sure that all sub cases of WebGL 2 conformance test can be passed in firefox nightly build (or in latest chromium), because the webgl 2.0 implementations are not mature, and visa versa. But we can continuously work on this. Finally, both conformance test suite for WebGL 2 and WebGL 2.0 Implementation in Chromium are all correct.

**Pull request merged**